test/framework/clusterstatevalidations/main_test.go

```diff
@@ -0,0 +1,94 @@
+package clusterstatevalidations_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/internal/test/envtest"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	clusterf "github.com/aws/eks-anywhere/test/framework/cluster"
+)
+
+var env *envtest.Environment
+
+const (
+	clusterName      = "test-cluster"
+	clusterNamespace = "test-namespace"
+)
+
+type clusterValidationTest struct {
+	t testing.TB
+	*WithT
+	config          clusterf.StateConfig
+	eksaSupportObjs []client.Object
+	otherObjs       []client.Object
+}
+
+func newStateValidatorTest(t testing.TB, clusterSpec *cluster.Spec) *clusterValidationTest {
+	tt := &clusterValidationTest{
+		t:     t,
+		WithT: NewWithT(t),
+		config: clusterf.StateConfig{
+			ClusterClient:           env.Client(),
+			ManagementClusterClient: env.Client(),
+			ClusterSpec:             clusterSpec,
+		},
+		eksaSupportObjs: []client.Object{
+			test.Namespace(clusterNamespace),
+			test.Namespace(constants.EksaSystemNamespace),
+			test.Namespace(constants.KubeSystemNamespace),
+		},
+		otherObjs: []client.Object{},
+	}
+	tt.t.Cleanup(tt.cleanup)
+	return tt
+}
+
+func (tt *clusterValidationTest) cleanup() {
+	ctx := context.Background()
+	apiExpecter := envtest.NewAPIExpecter(tt.t, tt.config.ClusterClient)
+	apiExpecter.DeleteAndWait(ctx, tt.allObjs()...)
+}
+
+func (tt *clusterValidationTest) addObjects(objs ...client.Object) {
+	tt.otherObjs = append(tt.otherObjs, objs...)
+}
+
+func (tt *clusterValidationTest) createAllObjects(ctx context.Context, excludedObjects ...client.Object) {
+	tt.t.Helper()
+	for _, obj := range tt.allObjs() {
+		isExcluded := false
+		for _, excluded := range excludedObjects {
+			isExcluded = equality.Semantic.DeepEqual(obj, excluded)
+			if isExcluded {
+				break
+			}
+		}
+		if !isExcluded {
+			envtest.CreateObjs(ctx, tt.t, tt.config.ManagementClusterClient, obj)
+		}
+	}
+}
+
+func (tt *clusterValidationTest) allObjs() []client.Object {
+	childObjects := tt.config.ClusterSpec.ChildObjects()
+	objs := make([]client.Object, 0, len(tt.eksaSupportObjs)+len(childObjects)+len(tt.otherObjs)+1)
+	objs = append(objs, tt.eksaSupportObjs...)
+	objs = append(objs, tt.config.ClusterSpec.Cluster)
+	objs = append(objs, tt.otherObjs...)
+	for _, o := range childObjects {
+		objs = append(objs, o)
+	}
+	return objs
+}
+
+func TestMain(m *testing.M) {
+	os.Exit(envtest.RunWithEnvironment(m, envtest.WithAssignment(&env)))
```

Comment:
do we need envtest for these tests? Can't we just use fakeclient?

Reply:
I'll see if we can do that instead!